Intermittent nil pointer dereference in collect()

Intermittently, we observe a nil pointer dereference in `typeCache.collect()`
This shows up in our logs after a few days:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x8856f2]
goroutine 363679 [running]:
v.io/v23/vom.(*TypeDecoder).Stop(0x0)
\u0009external/grailgo/vendor/v.io/v23/vom/type_decoder.go:109 +0x22
v.io/x/ref/runtime/internal/rpc.(*typeCache).collect(0xc4203cd400)
\u0009external/grailgo/vendor/v.io/x/ref/runtime/internal/rpc/typecache.go:115 +0x332
created by v.io/x/ref/runtime/internal/rpc.(*typeCache).writer
\u0009external/grailgo/vendor/v.io/x/ref/runtime/internal/rpc/typecache.go:46 +0x17a
+ rc-status
```

Cause appears to be a missing nil check in `typeCache.collect()` line 115.
Will submit a PR to fix.

--- a/x/ref/runtime/internal/rpc/typecache.go
+++ b/x/ref/runtime/internal/rpc/typecache.go
@@ -112,7 +112,9 @@ func (tc *typeCache) collect() {
 				if tce.cancel != nil {
 					tce.cancel()
 				}
-				tce.dec.Stop()
+				if tce.dec != nil {
+					tce.dec.Stop()
+				}
 				delete(tc.flows, conn)
 			}
 		}

